staging/src/k8s.io/apiserver/pkg/storage/etcd3/store.go

```diff
@@ -727,13 +729,20 @@ func (s *store) List(ctx context.Context, key string, opts storage.ListOptions,
 	var lastKey []byte
 	var hasMore bool
 	var getResp *clientv3.GetResponse
+	var numFetched int
+	var numEvald int
+	defer func() {
```

Comment:
I actually wouldn't make a defer here - we should only export this metric on success, not on all failures here.

I would actually move it to line 800 without defer.

Reply:
Since this is about costs, I think it is appropriate to record even for failures.

Reply:
OK - I buy that. Can you add a comment here along those lines?

Reply:
done